crates/bevy_ecs/src/query/state.rs

```diff
@@ -2229,6 +2230,7 @@ mod tests {
         let mut df = DefaultQueryFilters::empty();
         df.register_disabling_component(world.register_component::<Table>());
         world.insert_resource(df);
+        world.register_disabling_component::<IsResource>();
```

Comment:
I think we want to change this test, as well, so that we don't have to update it for every built-in disabling component.  It does want to reset the filters, but that can be done like

```rust
        let df = DefaultQueryFilters::from_world(&mut world);
        world.insert_resource(df);
        world.register_disabling_component::<Sparse>();
```

Reply:
Thanks for insisting on this, for the few PRs I do, there's a lot of fixing tests. I'm often lazy and just fix it in the simple ways, but working on testing robustness should help in the future.